Sources/xcproj/Referenceable.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  Referenceable.swift
+//  xcprojPackageDescription
+//
+//  Created by rmalik on 11/8/17.
+//
+
+import Foundation
+
+// Typealias that represents dictionary from string -> T where T is Referenceable (i.e. PBXObjects)
+
+public typealias ReferenceableCache<T: Referenceable & Equatable> = [String: T]
+
+public protocol Referenceable {
+    var reference: String { get }
+}
+
+extension Array where Element: Referenceable {
+
+    public var references: [String] {
+        return map { $0.reference }
+    }
+
+    public func contains(reference: String) -> Bool {
+        return contains { $0.reference == reference }
+    }
+
+    public func getReference(_ reference: String) -> Element? {
+        return first { $0.reference == reference }
+    }
+}
+
+
+extension Dictionary where Key == String, Value: Referenceable {
+    static func from(references: [Value]) -> Dictionary<String, Value> {
+        return Dictionary<String, Value>(uniqueKeysWithValues: references.flatMap { ($0.reference, $0) })
+    }
+
+    public var references: [String] {
+        return self.values.map { $0.reference }
+    }
+
+    public var referenceValues: [Value] {
```

Comment:
Is there much use in this?

Reply:
It's convenient instead of writing the return statement everywhere we need the list of values